config/src/lib.rs

```diff
@@ -1187,39 +1187,205 @@ impl Config {
+                    );
+                    toml_map.insert(&profile_str, value);
+                } else if let Some(non_inserted) = insert_profile(toml_map, &profile_str, value) {
+                    toml_map.insert(&profile_str, non_inserted);
+                } else {
+                    edited = true;
+                }
+            } else if profile == Config::DEFAULT_PROFILE &&
+                toml_map
+                    .get(Config::PROFILE_SECTION)
+                    .and_then(|val| val.as_table_like())
+                    .and_then(|table| table.get(&profile_str))
+                    .is_none() &&
+                insert_profile(
+                    toml_map,
+                    &profile_str,
+                    toml_edit::Item::Table(toml_edit::Table::new()),
+                )
+                .is_none()
+            {
+                // inserted default profile if it doesn't exist
+                edited = true;
+            }
+        }
+        // emit warnings for unknown sections
+        for section in
+            toml_doc.as_table().iter().map(|(k, _)| k).filter(|k| {
+                !(k == &Config::PROFILE_SECTION || Config::STANDALONE_SECTIONS.contains(k))
+            })
+        {
+            config_warn!("Unknown section [{}] found in TOML at {}. If you wish to use this section as a profile, run the command again with FOUNDRY_PROFILE=\"{}\".", section, toml_path.display(), section);
+        }
+        // write toml back to file
+        if edited {
+            if let Err(err) = fs::write(toml_path, toml_doc.to_string()) {
+                config_warn!("Could not write TOML to {}: {}", toml_path.display(), err);
+            }
+        }
+    }
+
+    /// Fix foundry.toml files. Making sure any implicit profile [<profile>] becomes
+    /// [profile.<profile>]
+    pub fn fix_tomls() {
+        let tomls = {
+            let mut tomls = vec![];
+            if let Some(global_toml) = Config::foundry_dir_toml().filter(|p| p.exists()) {
+                tomls.push(global_toml);
+            }
+            let local_toml = Env::var("FOUNDRY_CONFIG")
+                .map(PathBuf::from)
+                .unwrap_or_else(|| Config::default().__root.0.join(Config::FILE_NAME));
+            if local_toml.exists() {
+                tomls.push(local_toml);
+            } else {
+                config_warn!("No local TOML found to fix. Change the current directory to a project path or set the foundry.toml path with the FOUNDRY_CONFIG environment variable");
+            }
+            tomls
+        };
+        let profiles = {
+            let profile = Config::selected_profile();
+            let mut profiles = vec![Config::DEFAULT_PROFILE];
+            if profile != Config::DEFAULT_PROFILE {
+                profiles.push(profile);
+            }
+            profiles
+        };
+        for toml in tomls {
+            Self::fix_toml_for_profiles(toml, &profiles);
+        }
```

Comment:
does this only fix the selected and the default? others are ignored?

Reply:
yeah, my reasoning was that there could be other keys in the foundry.toml for other applications maybe (like how people use package.json) but at the same time, Cargo.toml will also emit warnings for unknown keys, so I guess we can just assume the foundry.toml is just for us...